cluster/kubernetes/kubernetes.go

```diff
@@ -402,3 +412,27 @@ func (c *Cluster) ImagesToFetch() registry.ImageCreds {
 
 	return allImageCreds
 }
+
+// getNamespaces returns a list of namespaces that the Flux instance is expected
+// to have access to and can look for resources inside of.
+// It returns a list of all namespaces unless a namespace whitelist has been set on the Cluster
+// instance, in which case it returns a list containing the namespaces from the whitelist
+// that exist in the cluster.
+func (c *Cluster) getNamespaces() ([]apiv1.Namespace, error) {
+	nsList := []apiv1.Namespace{}
+
+	namespaces, err := c.client.Namespaces().List(meta_v1.ListOptions{})
```

Comment:
@mwhittington21 Is this going to have the API call limiting effect you're after? It's better than asking for all the namespaces _then_ asking for all the things in them. Doing better again might be a tricky balance though, wdyt.

Reply:
I've already tested a couple of the commits on my 30k namespace cluster and the result is vastly improved. For starters, Flux isn't frozen due to the amount of API requests it has to make (~10 per namespace). The list operation will take a bit longer, but once it's done there are almost no follow up calls by comparison.

Big thumbs up from me for this current version improving things.